src/main/java/io/confluent/connect/hdfs/parquet/ParquetFileReader.java

```diff
@@ -18,49 +18,81 @@
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
 import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
 import org.apache.parquet.avro.AvroReadSupport;
 import org.apache.parquet.hadoop.ParquetReader;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Iterator;
 
 import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
 import io.confluent.connect.hdfs.SchemaFileReader;
 
-public class ParquetFileReader implements SchemaFileReader {
-
+public class ParquetFileReader implements SchemaFileReader,
+    io.confluent.connect.storage.format.SchemaFileReader<HdfsSinkConnectorConfig, Path> {
   private AvroData avroData;
 
   public ParquetFileReader(AvroData avroData) {
     this.avroData = avroData;
   }
 
   @Override
-  public Schema getSchema(Configuration conf, Path path) throws IOException {
+  public Schema getSchema(HdfsSinkConnectorConfig conf, Path path) {
     AvroReadSupport<GenericRecord> readSupport = new AvroReadSupport<>();
     ParquetReader.Builder<GenericRecord> builder = ParquetReader.builder(readSupport, path);
-    ParquetReader<GenericRecord> parquetReader = builder.withConf(conf).build();
-    GenericRecord record;
-    Schema schema = null;
-    while ((record = parquetReader.read()) != null) {
-      schema = avroData.toConnectSchema(record.getSchema());
+    try {
+      ParquetReader<GenericRecord> parquetReader = builder.withConf(conf.getHadoopConfiguration())
+          .build();
+      GenericRecord record;
+      Schema schema = null;
+      while ((record = parquetReader.read()) != null) {
+        schema = avroData.toConnectSchema(record.getSchema());
+      }
+      parquetReader.close();
+      return schema;
+    } catch (IOException e) {
+      throw new DataException(e);
     }
-    parquetReader.close();
-    return schema;
   }
 
   @Override
-  public Collection<Object> readData(Configuration conf, Path path) throws IOException {
+  public Collection<Object> readData(HdfsSinkConnectorConfig conf, Path path) {
     Collection<Object> result = new ArrayList<>();
     AvroReadSupport<GenericRecord> readSupport = new AvroReadSupport<>();
     ParquetReader.Builder<GenericRecord> builder = ParquetReader.builder(readSupport, path);
-    ParquetReader<GenericRecord> parquetReader = builder.withConf(conf).build();
-    GenericRecord record;
-    while ((record = parquetReader.read()) != null) {
-      result.add(record);
+    try {
+      ParquetReader<GenericRecord> parquetReader = builder.withConf(conf.getHadoopConfiguration())
+          .build();
+      GenericRecord record;
+      while ((record = parquetReader.read()) != null) {
+        result.add(record);
+      }
+      parquetReader.close();
+    } catch (IOException e) {
+      throw new DataException(e);
```

Comment:
Is `DataException` the right choice here? Couldn't this just be due to connectivity issues with HDFS? My specific concern is that this is not a retriable exception.